aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -883,7 +854,7 @@ void TensorIterator::compute_strides(const TensorIteratorConfig& config) {
       else
           op.stride_bytes.resize(ndim());
       for (size_t i = 0; i < original_shape.size(); i++) {
-        if (original_shape[i] == 1) {
+        if (original_shape[i] == 1 && shape_[offset + i] !=1) {
```

Comment:
This would be a great place to have a reference to a Note explaining the overall algorithm

Reply:
"strides of broadcasted dimensions are set to 0, and 0 compares equal to anything. Strides of not-broadcasted dimensions (including dimensions of size 1) participate in sorting."